exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/join/HashJoinBatch.java

```diff
@@ -532,6 +542,9 @@ public void close() {
     if (hashTable != null) {
       hashTable.clear();
     }
+
+    container.clear();
+    outputAllocator.close();
```

Comment:
Yes, it could be if we close before calling next()/buildSchema().  I will add a check for null.